caikit/runtime/client/remote_module_base.py

```diff
@@ -0,0 +1,496 @@
+            "timeout": self._connection.timeout,
+        }
+        # Send RPC request with or without streaming
+        if method.output_streaming:
+
+            def output_stream_parser():
+                """Helper function to stream result objects"""
+                for proto in service_rpc(grpc_request, **request_kwargs):
+                    yield response_dm_class.from_proto(proto)
+
+            # Attach reference of this RemoteModuleClass to the returned DataStream. This ensures
+            # the GRPC Channel won't get closed until after the DataStream has been cleaned up
+            return_stream = DataStream(output_stream_parser)
+            return_stream._source = self
+            return return_stream
+        else:
+            response = service_rpc(grpc_request, **request_kwargs)
+            return response_dm_class.from_proto(response)
+
+    @property
+    def _grpc_channel(self) -> grpc.Channel:
+        """Helper function to construct a GRPC channel
+        with correct credentials and TLS settings."""
+        # Short circuit if channel has already been set
+        if self.__grpc_channel:
+            return self.__grpc_channel
+
+        with self._channel_lock:
+            # Check for the channel again incase it was created during lock acquisition
+            if self.__grpc_channel:
+                return self.__grpc_channel
+
+            # Gather grpc configuration
+            target = self._get_remote_target()
+            options = list(self._connection.options.items())
+
+            # Generate secure channel
+            if self._tls.enabled:
+                grpc_credentials = grpc.ssl_channel_credentials(
+                    root_certificates=self._tls.ca_file_data,
+                    private_key=self._tls.key_file_data,
+                    certificate_chain=self._tls.cert_file_data,
+                )
+                channel = grpc.secure_channel(
+                    target, credentials=grpc_credentials, options=options
+                )
+            else:
+                channel = grpc.insecure_channel(target, options=options)
+
+            self.__grpc_channel = channel
+            return self.__grpc_channel
+
+    @staticmethod
+    def _close_grpc_channel(channel: grpc.Channel):
+        """Helper function to close a grpc channel. This should
+        be called on class deletion."""
+        channel.close()
+
+    ### Generic Helper Functions
+
+    def _get_remote_target(self) -> str:
+        """Get the current remote target"""
+        target_string = f"{self._connection.hostname}:{self._connection.port}"
+        if self._protocol == "grpc":
+            return target_string
+        else:
+            if self._tls.enabled:
+                return f"https://{target_string}"
+            else:
+                return f"http://{target_string}"
+
+    @staticmethod
+    def _get_streaming_arguments(**kwargs: Dict[str, Any]) -> List[str]:
```

Comment:
I think we have similar logic to this somewhere in the signature parsing to determine the shape of stream "frame" messages. Choosing which kwargs are streamable should probably use common code between client and server.

Reply:
hmmm, I looked around in Caikit Core and the DataModel classes and I couldn't find anything. Do you happen to have a link to the similar functions?